homeassistant/components/tplink/binary_sensor.py

```diff
@@ -0,0 +1,65 @@
+"""Support for TPLink binary sensors."""
+
+from __future__ import annotations
+
+from kasa import Feature, SmartDevice
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import TPLinkDataUpdateCoordinator
+from .entity import CoordinatedTPLinkEntity, _entities_for_device_and_its_children
+from .models import TPLinkData
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up sensors."""
+    data: TPLinkData = hass.data[DOMAIN][config_entry.entry_id]
+    parent_coordinator = data.parent_coordinator
+    device = parent_coordinator.device
+
+    entities = _entities_for_device_and_its_children(
+        device,
+        feature_type=Feature.BinarySensor,
+        entity_class=BinarySensor,
+        coordinator=parent_coordinator,
+    )
+
+    async_add_entities(entities)
+
+
+class BinarySensor(CoordinatedTPLinkEntity, BinarySensorEntity):
+    """Representation of a TPLink binary sensor."""
+
+    def __init__(
+        self,
+        device: SmartDevice,
+        coordinator: TPLinkDataUpdateCoordinator,
+        feature: Feature,
+        parent: SmartDevice = None,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(device, coordinator, feature=feature, parent=parent)
+        # TODO: generalize creation of entitydescription into CoordinatedTPLinkEntity?
+        self.entity_description = BinarySensorEntityDescription(
+            key=feature.id,
+            translation_key=feature.id,
+            name=feature.name,
+            icon=feature.icon,
+            **feature.hass_compat.dict(),
```

Comment:
Its a bit strange for the library to know about HomeAssistant.

I think we should rename this `legacy_compat` instead

Reply:
So this is a counterpart for https://github.com/python-kasa/python-kasa/pull/879 - basically, we need some way to inform homeassistant on its own special handling for some features. An alternative would be to store this information inside the integration, but I was thinking that it would be better to keep this local.

We could inverse it by 1) adding static ids inside python-kasa and 2) using those ids to map for wanted parameters. Do you think that would be a better approach?

Reply:
We definitely shouldn't be teaching the library about Home Assistant.  We should store this information in the integration.

I think its fine to add information to `python-kasa` to make it easier to map in Home Assistant, but its separation of concerns problem for the library to start learning about HA

Reply:
https://github.com/python-kasa/python-kasa/pull/904 requires static identifiers, so we can move the mapping inside the integration.